src/map/latent_effect_container.cpp

```diff
@@ -1674,7 +1675,23 @@ void CLatentEffectContainer::CheckLatentsZone()
             }
             break;
         case LATENT_WEATHER_ELEMENT:
-            if (zoneutils::GetWeatherElement(zoneutils::GetZone(m_POwner->getZone())->GetWeather()) == m_LatentEffectList.at(i)->GetConditionsValue())
+            // In the default case we need to pull weather from zone/player
+            if (weather == MAX_WEATHER_ID)
```

Comment:
It is a default condition - to be used where CheckLatentsZone() is called without an element.  The only method that should be passing an element would be CZoneEntities::WeatherChange(...).  Basically we need a way to inform the latent checking of a potentially new weather.  At the time of checking we only have access to the old weather for the zone so it would be an incorrect check.